_drafts/2016-06-03-Building-a-comfy-frontend-stack.md

```diff
@@ -0,0 +1,212 @@
+from routing to HTTP requests for you, with detailed documentation. However,
+for some of the more volatile features, Marques found the documentation
+confusing, incomplete, and inconsistent. This'll probably cease to be an issue
+once Angular 2 is released properly.
+
+Angular also offers some innovative ways of managing your data with the
+observables mechanic. They basically work like Unix pipes and let you chain
+operations together and subscribe to changes anywhere in the pipeline, then
+respond to those changes in your application. Angular is also well tested and
+the community has many examples, though some may be outdated (which is a problem
+for all three of the options we're looking at today).
+
+There are some disadvantages to Angular. Though Google has promised improvements
+to this, Angular 2 is pretty slow. There are lots of debugging things going on
+and the startup times are rough. The transition from Angular 1 to Angular 2 is
+also very jarring and those with preconceptions about how Angular works are
+going to have a hard time. Some people also criticize the "do everything"
+approach that Angular takes, preferring the Unix philosophy where tools and
+libraries with a smaller scope are composed to fill the full picture.
+
+## Mithril & ES5
+
+Mithril is for those who took the lessons of the past few years of JavaScript
+and concluded that not much needed to change. Abe quickly realized that Mithril
+was not the right call, but he soldiered on to be absolutely sure and see if
+there were any lessons to be learned that could be applied to our eventual
+choice. Abe had some praise for it, though. One of the big positives to Mithril
+is that it's very quick to get up and running. There's no transpiling or
+boilerplate to set up, no webpack or gulp, no package.json, or anything like
+that. You can just drop it into your page and get started like any old school
+JavaScript library. It's also pretty simple for developers who are comfortable
+with ES5 libraries and patterns to understand and use.
+
+Abe found that the deeper he got into it, though, the less effective it was. The
+documentation is lacking and it handles more complex situations poorly. He also
+called some criticism on its separation of concerns, noting that idiomatic
+Mithril models are aware of the DOM. ES5 is also a pretty bad language, which
+we've all known for a long time. Transpiling is the right call for your sanity.
+
+I also asked Abe about the MVC pattern in general, and he still feels like it's
+the right call in some situations. Abe worries that once your application gets
+big enough and cross dependencies start to pile up, MVC falls apart. [Facebook
+feels the same way](http://facebook.github.io/flux/) (though I think the
+prevailing thought is that Flux isn't quite right, either).
+
+## React.js & ES6
+
+React.js is Facebook's offering for the new age of JavaScript, and Redux is a
```

Comment:
it's not necessarily that React/Redox are "based" on them - these technologies just existed before and provided similar ways of organization